### PR TITLE
Trace backing store events

### PR DESCRIPTION
**Pull request by jasagredo**
# Description

Do this in consensus instead of polluting the tracing code too much with low level stuff. 

Part of #227

### CODE REVIEW

**Review by jorisdral — CHANGES_REQUESTED** [review]

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs [hunk 1]
```diff
@@ -199,3 +189,35 @@ instance Show StoreDirIsIncompatible where
...
+
+data TVarTraceEvent =
+    TVarTraceOpening
+  | TVarTraceOpened
+  | TVarTraceClosing
+  | TVarTraceClosed
+  | TVarTraceCopying                  !FS.FsPath -- ^ To
+  | TVarTraceCopied                   !FS.FsPath -- ^ To
+  | TVarTraceInitialisingFromSnapshot !FS.FsPath
+  | TVarTraceInitialisedFromSnapshot  !FS.FsPath
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs]: Done

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs [hunk 2]
```diff
@@ -199,3 +189,35 @@ instance Show StoreDirIsIncompatible where
...
+  | TVarTraceWrite   !(WithOrigin SlotNo) !SlotNo
+  | TVarTraceInitialisingFromValues !(WithOrigin SlotNo)
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs]: Done

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs [hunk 3]
```diff
@@ -199,3 +189,35 @@ instance Show StoreDirIsIncompatible where
     <> show p
     <> ".\nPre-UTxO-HD and LMDB implementations are incompatible with the In-Memory \
        \ implementation. Please delete your ledger database directory."
+
+{-------------------------------------------------------------------------------
+  Tracing
+-------------------------------------------------------------------------------}
+
+data TVarTraceEvent =
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs]: Done

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs [hunk 4]
```diff
@@ -199,3 +189,35 @@ instance Show StoreDirIsIncompatible where
...
+showTrace TVarTraceOpening = "Opening InMemory backing store"
+showTrace TVarTraceOpened  = "Opened InMemory backing store"
+showTrace TVarTraceClosing = "Closing InMemory backing store"
+showTrace TVarTraceClosed  = "Closed InMemory backing store"
+showTrace (TVarTraceCopying p) = "Copying InMemory backing store to path " <> showT p
+showTrace (TVarTraceCopied p) = "Copied InMemory backing store to path " <> showT p
+showTrace (TVarTraceInitialisingFromSnapshot p) = "Initialising InMemory backing store from snapshot at " <> showT p
+showTrace (TVarTraceInitialisedFromSnapshot p) = "Initialised InMemory backing store from snapshot at " <> showT p
+showTrace (TVarTraceInitialisingFromValues s) = "Initialising InMemory backing store from values at slot " <> showT s
+showTrace (TVarTraceWrite sl1 sl2) = "Wrote to InMemory backing store up to slot " <> showT sl2 <> " onto " <> showT sl1
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/InMemory.hs]: Done

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/LMDB.hs [hunk 5]
```diff
@@ -658,6 +662,33 @@ data TraceValueHandle
...
+showT = pack . show
+
+showTrace :: TraceLMDB -> Text
+showTrace TDBOpening     = "Opening LMDB"
+showTrace (TDBOpened p)  = "Opened LMDB at " <> showT p
+showTrace (TDBClosing p) = "Closing LMDB at " <> showT p
+showTrace (TDBClosed p)  = "Closed LMDB at " <> showT p
+showTrace (TDBCopying p1 p2) = "Copying LMDB from " <> showT p1 <> " to " <> showT p2
+showTrace (TDBCopied p1 p2)  = "Copied LMDB from " <> showT p1 <> " to " <> showT p2
+showTrace (TDBWrite sl1 sl2)  = "Wrote to LMDB up to slot " <> showT sl2 <> " onto " <> showT sl1
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/BackingStore/LMDB.hs]: Done

**Review by jorisdral — APPROVED** [review]

**Pull request merged**